### ISSUE TITLE
/etc/rc.local fails because unmounting /lib/modules will fail when more then one chroot is running.

### ISSUE THREAD

**Issue by divx118** [divx118/crouton-packages#15]
We need to be more specific when unmounting /lib/modules. It must only unmount the bind mounts of the current chroot.

**Comment by shounak-de** [comment]
Is this still open? I think this bit me when trying the restore a few days ago:
`sudo sh ~/Downloads/change-kernel-flags -r`
I got a OS missing or damaged (ctrl-d did not work) 
I eventually had to do a full restore via USB from another computer

My chromebook was a toshiba chromebook2 (swanky)

**Comment by divx118** [comment]
Well this specific issue shouldn't cause that. It could be that you restored an older kernel from previous chromeos updates. I need to add a warning for that. However normally if the restored kernel is failing to boot it will switch to the other kernel. So this is a bit strange, but not impossible that you indeed need to restore with USB. It will always be a bit risky, I will take another look at the code to see if I can make it even more safe.

BTW sorry to hear you had todo a full restore.

**Comment by shounak-de** [comment]
@divx118 You are right, it may not be related.

BTW, I do not think it was an older kernel though that caused the issue: Here are the steps that caused it to bork:

[1] On clean dev mode with one chroot env installed, (but not active/running) I ran:
`sudo sh ~/Downloads/change-kernel-flags`

[2] I did not run `sudo sh setup-headers.sh` at all: I realized that after changing the kernel flags, the google voice search was not working on chromeos. My kid uses the "Ok Google" searches quite frequently, so I thought maybe restoring the kernel would bring that back, so I re-ran the change kernel flags script with a `-r`

[3] It completed successfully (found the backed up kernel, and restored) and asked be to reboot.

That that point I was stuck with the OS missing or damaged screen. No biggie, I had a backup of the chrooted environment, so I just restored from there without issues. My question to you would be do you know that changing the kernal flags  would indeed limit the chromeOS voice input feature?

**Comment by divx118** [comment]
>My question to you would be do you know that changing the kernal flags would indeed limit the chromeOS voice input feature?

No, changing the kernel flags to allow module loading wouldn't interfere with any other chromeos features like google voice.
